peers/app_request_network.go

```diff
@@ -257,3 +270,62 @@ func (n *AppRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) set.Set[id
 
 	return trackedNodes
 }
+
+// Helper struct to hold connected validator information
+// Warp Validators sharing the same BLS key may consist of multiple nodes,
+// so we need to track the node ID to validator index mapping
+type ConnectedCanonicalValidators struct {
```

Comment:
I think we should move this struct to a separate file, and ideally have mockign tests

Reply:
Eh, I don't think it's worth all that special treatment since this type is POD. It's really just there to simplify calls to `ConnectToCanonicalValidators` and keep associated data packaged together.